pkg/kubectl/cmd/run.go

```diff
@@ -336,3 +302,110 @@ func getRestartPolicy(cmd *cobra.Command, interactive bool) (api.RestartPolicy,
 		return "", cmdutil.UsageError(cmd, fmt.Sprintf("invalid restart policy: %s", restart))
 	}
 }
+
+func generateService(f *cmdutil.Factory, cmd *cobra.Command, args []string, serviceGenerator string, paramsIn map[string]interface{}, namespace string, out io.Writer) error {
+	generator, found := f.Generator(serviceGenerator)
+	if !found {
+		return fmt.Errorf("missing service generator: %s", serviceGenerator)
+	}
+	names := generator.ParamNames()
+
+	port := cmdutil.GetFlagInt(cmd, "port")
```

Comment:
Should we update the description of `--port` since it can be container port or service port?